### PR TITLE
PHOENIX-6208 Backport the assembly changes in PHOENIX-6178 to 4.x

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  5s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 43s |  4.x passed  |
| -1 :x: |  compile  |   0m 11s |  phoenix-assembly in 4.x failed.  |
| -1 :x: |  javadoc  |   0m  9s |  phoenix-assembly in 4.x failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 17s |  the patch passed  |
| -1 :x: |  compile  |   0m  8s |  phoenix-assembly in the patch failed.  |
| -1 :x: |  javac  |   0m  8s |  phoenix-assembly in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m  8s |  phoenix-assembly in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m  8s |  phoenix-assembly in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  18m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/941 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux e79abcb9eb6a 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / ac0538b |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/branch-compile-phoenix-assembly.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/branch-javadoc-phoenix-assembly.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/patch-compile-phoenix-assembly.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/patch-compile-phoenix-assembly.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/patch-javadoc-phoenix-assembly.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/artifact/yetus-general-check/output/patch-unit-phoenix-assembly.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/testReport/ |
| Max. process+thread count | 76 (vs. ulimit of 30000) |
| modules | C: phoenix-assembly U: phoenix-assembly |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-941/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by joshelser** [comment]
@stoty  can you summarize what these changes do? I acknowledge that these are a subset of the bigger release-creation steps, but it's not clear what benefit these changes have.

**Comment by stoty** [comment]
These changes are necessary to work with new release script in master. The release script in master is meant to be used with all subprojects and branches, including 4.x.

- Dropped the source assembly, which was never used, the new scripts use the tarred source repo instead as the source release artifact.
- Instead of assembling a separate binary release tarball in the make_rc.sh script, we now use the tarball that phoenix-assembly creates. I have also add -bin to the name of the assembly tarball to conform to the naming conventions, and added the binary license files that we were using in the old release tarball.
- Removed the leftover references to the python driver that was moved to the PQS repo (unrelated cleanup)

**Review by joshelser — APPROVED** [review]
Thanks, Istvan! Looks great.
